### PR TITLE
Add can_read to memories

### PR DESCRIPTION
**Pull request by yamaguchi1024**
I have no idea why we didn't discover this until now...

### CODE REVIEW

**Comment by yamaguchi1024** [comment]
FYI this is coming from Adrian's email

**Comment by alexreinking** [comment]
I think the problem is with this line:

https://github.com/exo-lang/exo/blob/6e7419ee9642ed92993a3403b7bde7e85b24c701/src/exo/memory.py#L95

Should be `return`, not `raise`... what issue was Adrian seeing? I'm not sure I got the email.

**Comment by yamaguchi1024** [comment]
@alexreinking The line is fine, it's trying to say that whoever inherits this class should specify if the memory is readable or not, which I think is fair.

File: tests/test_neon.py [hunk 1]
```diff
@@ -9,10 +9,22 @@
 from exo import proc
 from exo.platforms.neon import *
 from exo.stdlib.scheduling import *
+from exo.memory import MemGenError
 
 import numpy as np
 
 
+@pytest.mark.isa("neon")
```
Comment (alexreinking) [inline tests/test_neon.py:17]: This test doesn't actually run anything on NEON, it just needs to get as far as generating C. Therefore it should never be skipped and you can check that `read_neon.c_code_str()` throws the expected `MemGenError`.

**Review by alexreinking — CHANGES_REQUESTED** [review]
No reason to skip this test.

File: tests/test_neon.py (hunk 1 above)
Comment (yamaguchi1024) [reply tests/test_neon.py:17]: That makes sense!

**Review by alexreinking — APPROVED** [review]
Looks good when green

**Pull request merged**